### PR TITLE
Fix issue 1769

### PR DESCRIPTION
**Pull request by danilat**
This is the explanation of the bug I found https://github.com/strongloop/loopback/issues/1769. In my PR I have a ugly patch in this commit https://github.com/danilat/loopback/commit/e028b56aad42b0828a9bdd10dcaa9e8892344c34.

Now I'm working in a workaround with some changes in connector-remote parsing all the valid date-format strings in the respose objects. What do you think about this?

### CODE REVIEW

**Comment by slnode** [comment]
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

**Comment by superkhau** [comment]
Can you fix the PR so they pass CI?

**Comment by danilat** [comment]
Fixed

File: common/models/access-token.js [hunk 1]
```diff
@@ -132,26 +137,26 @@ module.exports = function(AccessToken) {
    * @param {Boolean} isValid
    */
 
-  AccessToken.prototype.validate = function(cb) {
+  AccessToken.validate = function(token, cb) {
```
Comment (bajtos) [inline common/models/access-token.js]: This is a breaking change, we need to add a backwards-compatible wrapper method:

```js
AccessToken.prototype.validate = function(cb) {
  AccessToken.validate(this, cb); // or this.toObject()?
};
```

Comment (bajtos) [inline common/models/access-token.js]: @danilat pardon me if my answer is obvious, but why is it necessary to change this method from a prototype method to a static method?

File: common/models/access-token.js [hunk 2]
```diff
@@ -101,7 +101,12 @@ module.exports = function(AccessToken) {
         if (err) {
           cb(err);
         } else if (token) {
-          token.validate(function(err, isValid) {
+          var possibleDate = new Date(token.created);
+          var isAValidDate = !isNaN(possibleDate.getTime());
+          if (isAValidDate) {
+            token.created = new Date(token.created);
+          }
```
Comment (bajtos) [inline common/models/access-token.js]: This looks suspicious. What part of the system is producing `token.created` that is not already a `Date` instance? I'd rather fix the problem at the source.

/cc @ritch @kraman

Comment (danilat) [inline common/models/access-token.js]: I have an application using connector-remote datasource for AccessToken, this was a temporal fix but I fixed the source in the connector module https://github.com/danilat/loopback-connector-remote/commit/db9446705dedabea6811a68faf4ce33e48a93beb

File: common/models/access-token.js (hunk 1 above)
Comment (danilat) [inline common/models/access-token.js]: In the connector-remote ds scenario for AccessToken model the prototype method doesn't work, changing the implementation to static I was the solution that I found, I tried a couple of workarounds but nothing of them worked. I'm sure that a more elegant implementation could be done.

**Comment by bajtos** [comment]
I am afraid this is fixing the wrong place. Ideally, loopback-connector-remote should support remote methods and seamlessly convert (coerce) response properties to the correct type. See my https://github.com/strongloop/loopback/issues/1769#issuecomment-265717471

I am closing this pull request as rejected.

**Pull request closed without merging**